teuthology/provision/openstack.py

```diff
@@ -57,52 +57,78 @@ def init_user_data(self, os_type, os_version):
         log.debug("user_data file contents:")
         log.debug(user_data)
 
+    def _create_volume(self, volume_name, size):
```

Comment:
I understand that this is just a refactoring and this method is split away from the _attach_volume_, but it brings the defect when there is a volume with the same name and owned by the same teuthology server, which is left from previous executions, for example got stuck at some status, like attaching or something, and can't be removed due to openstack issues.

I would suggest to make this method to not check existence of a volume, or probably just WARN the user that it exists, and always make an attempt to create volume and return its *ID* which will be used for attaching to target lately instead of its _Name_.